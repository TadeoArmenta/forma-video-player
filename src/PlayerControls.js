import React, { Component } from 'react';
import {
  View
} from 'react-native';
import PropTypes from 'prop-types';

import Header from './Header'
import Body from './Body'
import Footer from './Footer'
import {ProgressBar} from './Controls'
import layout from './layout'
import styles from './styles'

export class PlayerControls extends Component {

  constructor(props) {
    super(props)
  }

  render() {
    const {Header, Body, Footer} = this.props.layout
    const {container, progressBar} = this.props.styles

    return (
      <View style={container}>
        {Header}
        {Body}
        {Footer}
        <View style={progressBar}>
          <ProgressBar />
        </View>
      </View>
    )
  }

}

PlayerControls.defaultStyles = {
  playerControls: {
    flex: 1
  },
  progressBar: {position: 'absolute', bottom: 30, left: 10, right: 10, height: 40}
}

export const LayoutWrapped = layout({
  Header: <Header />,
  Body: <Body />,
  Footer: <Footer />
})(PlayerControls)

export const Styled = styles((styles, theme) => ({
  container: {
    flex: 1
  },
  progressBar: {
    position: 'absolute',
    bottom: 30,
    left: 10,
    right: 10,
    height: 40
  }
}))(LayoutWrapped)

export default Styled
