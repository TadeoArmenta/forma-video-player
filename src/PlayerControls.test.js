import React, {Component} from 'react'
import ShallowTestRenderer from 'react-test-renderer/shallow'

import {PlayerControls} from './PlayerControls'

describe('PlayerControls', () => {

  class TestComponent {render() {return null}}

  let renderer

  beforeEach(() => {
    renderer = new ShallowTestRenderer()
  })

  it('should render layout', () => {
    const layout = {
      Header: <TestComponent />,
      Body: <TestComponent />,
      Footer: <TestComponent />,
    }
    const rendered = renderer.render(<PlayerControls layout={layout} styles={{}} />)

    expect(rendered).toMatchSnapshot()
  })

})
