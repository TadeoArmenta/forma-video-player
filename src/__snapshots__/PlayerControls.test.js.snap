// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerControls should render layout 1`] = `
<View
  style={undefined}
>
  <TestComponent />
  <TestComponent />
  <TestComponent />
  <View
    style={undefined}
  >
    <Connect(_class) />
  </View>
</View>
`;
